sunpy/map/mapcube.py

```diff
@@ -317,14 +318,40 @@ def all_maps_same_shape(self):
         """
         return np.all([m.data.shape == self.maps[0].data.shape for m in self.maps])
 
+    def at_least_one_map_has_mask(self):
+        """
+        Tests if at least one map has a mask.
+        """
+        return np.any([m.mask is not None for m in self.maps])
+
+    def no_map_has_mask(self):
```

Comment:
Yes, I believe you are right.  You just want to know if there is at least
one mask or not.  Thanks - this simplifies the code even more.

On Tue, Feb 9, 2016 at 10:17 AM, Stuart Mumford <notifications@github.com>
wrote:

> In sunpy/map/mapcube.py
> <https://github.com/sunpy/sunpy/pull/1660#discussion_r52320576>:
>
> > @@ -317,14 +318,40 @@ def all_maps_same_shape(self):
> >          """
> >          return np.all([m.data.shape == self.maps[0].data.shape for m in self.maps])
> >
> > +    def at_least_one_map_has_mask(self):
> > +        """
> > +        Tests if at least one map has a mask.
> > +        """
> > +        return np.any([m.mask is not None for m in self.maps])
> > +
> > +    def no_map_has_mask(self):
>
> is this the same as not self.at_least_one_map_has_mask()??
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sunpy/sunpy/pull/1660/files#r52320576>.
>